### PR TITLE
Made it possible for totalCount to be a promise

### PR DESCRIPTION
**Pull request by rsolyanik**
An error occurs when trying to return a promise to totalCount
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documented?   | no
| Fixed tickets | #286 
| License       | MIT

### CODE REVIEW

File: src/Relay/Connection/ConnectionInterface.php [hunk 1]
```diff
@@ -35,12 +35,14 @@ public function setPageInfo(PageInfoInterface $pageInfo): void;
     /**
      * Get the total count.
      *
-     * @return int
+     * @return int|object|null
      */
```
Comment (murtukov) [inline src/Relay/Connection/ConnectionInterface.php:39]: Need to specify the exact returned object class or interface (promise in this case, I guess). And updating the description would be nice too.

File: src/Relay/Connection/ConnectionInterface.php [hunk 2]
```diff
@@ -35,12 +35,14 @@ public function setPageInfo(PageInfoInterface $pageInfo): void;
...
-    public function getTotalCount(): ?int;
+    public function getTotalCount();
 
     /**
      * Set the total count.
+     *
+     * @param int|object $totalCount
```
Comment (murtukov) [inline src/Relay/Connection/ConnectionInterface.php:45]: Same here

File: src/Relay/Connection/Output/Connection.php [hunk 3]
```diff
@@ -16,7 +16,9 @@ class Connection implements ConnectionInterface
     protected array $edges;
 
     protected ?PageInfoInterface $pageInfo;
-    protected ?int $totalCount = null;
+
+    /** @var int|object|null */
+    protected $totalCount = null;
```
Comment (murtukov) [inline src/Relay/Connection/Output/Connection.php:21]: Same here

File: src/Relay/Connection/Output/Connection.php [hunk 4]
```diff
@@ -16,7 +17,9 @@ class Connection implements ConnectionInterface
     protected array $edges;
 
     protected ?PageInfoInterface $pageInfo;
-    protected ?int $totalCount = null;
+
+    /** @var int|Promise|null Total count or promise that returns the total count */
```
Comment (murtukov) [inline src/Relay/Connection/Output/Connection.php:21]:
```suggestion
    /** 
     * Total count or promise that returns the total count.
     * @var int|Promise|null
     */
```

**Comment by murtukov** [comment]
ping @mcg-web 

Could you review this PR? I don't work with Relay, so I can't say if it's a good thing to allow to return a promise from the getter.

**Comment by murtukov** [comment]
@rsolyanik before pushing any changes you have to execute `composer fix-cs` to bring the code to predefined coding styles and `composer static-analysis` to check if there are any problems. If there are any problems, Travis-CI check won't pass.

**Comment by rsolyanik** [comment]
> ping @mcg-web
> 
> Could you review this PR? I don't work with Relay, so I can't say if it's a good thing to allow to return a promise from the getter.
This is necessary to load total counts via data loader in nested paged lists
Something like 
`{
  customers(first: 1000) {
    edges {
      node {
        id,
        loyaltyAccounts {
          totalCount
          edges {
            node {
              ....
            }
          }
        }
      }
    }
  }
}`

**Review by murtukov — APPROVED** [review]

**Pull request merged**